src/ssh/cockpitsshrelay.c

```diff
@@ -845,183 +845,28 @@ do_password_auth (CockpitSshData *data)
   return rc;
 }
 
-struct CockpitSshPromptData {
-  CockpitSshData *data;
-  const gchar *identity;
-  gboolean did_prompt;
-};
-
-/* We don't support unlocking identities within cockpit-ssh so fail here */
 static int
-prompt_for_identity_password (const char *prompt, char *buf, size_t len,
-                              int echo, int verify, void *userdata)
+intercept_prompt (const char *prompt, char *buf, size_t len,
+                  int echo, int verify, void *userdata)
 {
-  struct CockpitSshPromptData *prompt_data = userdata;
-  prompt_data->data->problem_error = g_strdup_printf ("locked identity: %s", prompt_data->identity);
-  prompt_data->did_prompt = TRUE;
+  CockpitSshData *data = userdata;
+  char *identity = NULL;
+  if (ssh_userauth_publickey_auto_get_current_identity (data->session, &identity) == SSH_OK)
+    {
+      data->problem_error = g_strdup_printf ("locked identity: %s", identity);
```

Comment:
As a heads up, the exact "locked identity" string here is important, see pkg/shell/machines/machine-dialogs.js